File: pacade/kotlin/compiler/src/main/kotlin/lines/compiler.kt
package lines

import lines.builder.CompilerBuilder
import lines.compiler.JavaCompiler
import lines.compiler.Node
import lines.compiler.Parser
import lines.compiler.Scanner
import lines.domain.Compiler

fun main() {
    val compiler : Compiler = JavaCompiler(CompilerBuilder(
        Node(),
        Parser(),
        Scanner(),
    ))

    compiler.Compile()
}

File: pacade/go/compiler/app/compiler/compiler.go
package compiler

import (
	"compiler/app/builder"
	"compiler/domain"
)

type Compiler struct {
	element *builder.CompilerElement
}

func (c Compiler) Compile() {
	c.element.GetParser().Process()

	c.element.GetScanner().Process()

	c.element.GetProgramNode().Process()
}

func NewCompiler(element *builder.CompilerElement) domain.Compiler {
	return &Compiler{
		element: element,
	}
}
